openupgradelib/openupgrade.py

```diff
@@ -836,6 +867,26 @@ def map_values(
             "map_values is called with the same value for source and old"
             " columns : %s",
             source_column)
+    if check_completeness:
+        values = {
+            'table': table,
+            'source_column': source_column,
+        }
+        query = ("SELECT {source_column!s} FROM {table!s}").format(**values)
+        cr.execute(query)
+
+        set1 = set([i[0] for i in cr.fetchall()])
+        set2 = set([i[0] for i in mapping])
+        missing = set1 - set2
+
+        if (missing - set([None])):
+            for dbid in missing:
+                logger.error((
+                    "'map_values_by_xml' has detected a missing mapping for "
+                    "Database ID {0} in Source Column (check_completness "
+                    "enabled)."
+                ).format(dbid))
+            raise
```

Comment:
I'm not sure about this as it's per line level, so raising would raise on every missing mapping, so you would need to run the script x times in order to get all x faulty mappings, instead of a simple look at the logs.